api/v1alpha6/openstackmachine_conversion.go

```diff
@@ -342,7 +385,16 @@ func Convert_v1beta1_OpenStackMachineSpec_To_v1alpha6_OpenStackMachineSpec(in *i
 
 /* OpenStackMachineStatus */
 
+func restorev1alpha6MachineStatus(previous *OpenStackMachineStatus, dst *OpenStackMachineStatus) {
+	if dst.InstanceState == nil || *dst.InstanceState == "" {
+		dst.InstanceState = previous.InstanceState
```

Comment:
the other restore functions skip the setting of previous into dst when the resource is empty/nil. Should we follow the same pattern here?

Reply:
I've been adding that incrementally. As this is a new function I should have added it here.